DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Sheets.v4;
+using Google.Apis.Sheets.v4.Data;
+using static Google.Apis.Sheets.v4.SpreadsheetsResource.ValuesResource;
+
+namespace DiscordBot.DataAccess
+{
+    public class EventsSheetsService
+    {
+        private static readonly string[] scopes = { SheetsService.Scope.Spreadsheets };
+        private static readonly string applicationName = "Softwire Discord Bot";
+        private static readonly string spreadsheetId = "";
+
+        private SheetsService sheetsService;
+
+        private static async Task<ServiceAccountCredential> GetCredentialsAsync(string path = "credentials.json")
+        {
+            await using var stream =
+                new FileStream(path, FileMode.Open, FileAccess.Read);
+
+            var googleCredential = GoogleCredential.FromStreamAsync(stream, CancellationToken.None);
+
+            return (await googleCredential)
+                .CreateScoped(scopes)
+                .UnderlyingCredential as ServiceAccountCredential;
+        }
+
+        public EventsSheetsService() { }
+
+        public async Task StartService()
```

Comment:
Is there a reason not to do this in the constructor? 🙂

Reply:
The reason I'm doing this outside the constructor is that starting the service needs to run some asynchronous code. I suppose I could just block on it because the bot can't start running until this service has been set up 🙂

Reply:
Fair enough, I think a StartService method makes sense in that case 🙂

Reply:
Done this in 009105c 🙂